### PR TITLE
DEV-432 Document how to get what bib rights would be for an item curr…

### PR DESCRIPTION
**Pull request by moseshll**
…ently

- Add cgi/BibRights.pm module for wrangling MARC, Bib API, and bib_rights.
- Add cgi/bib_rights.tt UI for in-CRMS queries.
- Add bin/bib_rights.pl script for querying one or more HT ids and/or catalog ids.
- Added module's test coverage at 100%.
- Coveralls support via GitHub action.
- Metadata module (Bib API wrapper) can use local fixtures when running tests.
- Drastically reduce the verbosity of bib_rights tests with Capture::Tiny.
- Proudly display our dismal test coverage stats on README.md.
- Add scripts in bib_rights to t/bin.t sanity tests.
- Remove MYSQL_ROOT_PASSWORD from test docker-compose.
- Suppress bib_rights.pm STDERR noise in rights import script.

### CODE REVIEW

File: cgi/BibRights.pm [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+# Returns a structure {entries => [bib_rights_info+], optional error => errstr}
+sub query {
+  my $self = shift;
+  my $id   = shift;
+
+  my $br;
+  # bib_rights.pm likes to emit cutoff year debugging info to STDERR.
+  # So suppress it.
+  my ($stderr, @result) = Capture::Tiny::capture_stderr sub {
```
Comment (aelkiss) [inline cgi/BibRights.pm:40]: Yuck. Worth an update to bib_rights.pm?

File: docker-compose.test.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+version: '3'
+
+services:
+
+  mariadb:
+    build: docker/db
+    restart: always
+    environment:
+      MYSQL_ROOT_PASSWORD: 'crms'
+    ports:
```
Comment (aelkiss) [inline docker-compose.test.yml:10]: Is there a reason to expose mysql when running tests?

File: cgi/BibRights.pm (hunk 1 above)
Comment (moseshll) [reply cgi/BibRights.pm:40]: This makes it so I don't *have* to wade into PZP/bib_rights right now... but yes. There are several actions I would like to take in there wrt print to STDERR before long.

**Review by aelkiss — COMMENTED** [review]
Haven't tried it, but generally looks fine to me. I would rather see us just clean up the bib_rights stuff (e.g. by adding a DEBUG flag) than work around it here, but maybe there are reasons why that's inadvisable.

File: cgi/BibRights.pm (hunk 1 above)
Comment (aelkiss) [reply cgi/BibRights.pm:40]: Let's add a task to the post-zephir processing epic then

File: docker-compose.test.yml (hunk 2 above)
Comment (moseshll) [reply docker-compose.test.yml:10]: Worth trying without it --s econd-generation copy pasta. The original docker-compose has it for ease of use in trying schema changes (I think) but not needed here.

File: cgi/BibRights.pm (hunk 1 above)
Comment (moseshll) [reply cgi/BibRights.pm:40]: DEV-456

File: docker-compose.test.yml (hunk 2 above)
Comment (moseshll) [reply docker-compose.test.yml:10]: Tests failed on `wait-for` (I think) with empty environment, using MYSQL_RANDOM_ROOT_PASSWORD subsequently allowed the tests to pass, so keeping that. A bit weird.

**Comment by moseshll** [comment]
@aelkiss This branch is on dev-2. Example uses:
```
/htapps/moseshll.babel/crms/bin/bib_rights.pl aeu.ark:/13960/t0000j18q
```
For dump of bib_rights_info fields
```
/htapps/moseshll.babel/crms/bin/bib_rights.pl -v aeu.ark:/13960/t0000j18q
```
Use with catalog id
```
/htapps/moseshll.babel/crms/bin/bib_rights.pl 100251659
```

**Comment by aelkiss** [comment]
I needed to set `SDRROOT=/htapps/moseshll.babel` and then it was otherwise fine.

`bib_rights.pm` provoked a warning:

```
aelkiss@macc-ht-web-dev-169:/htapps/moseshll.babel/crms$ perl -w bin/bib_rights.pl mdp.39015076257560
"my" variable $field masks earlier declaration in same scope at /htapps/moseshll.babel/crms/post_zephir_processing/bib_rights.pm line 341.
mdp.39015076257560 pdus
```
I don't think that's in scope for this PR, but I will take a look at what's going on there.

**Comment by aelkiss** [comment]
https://github.com/hathitrust/post_zephir_processing/pull/20/commits/f867c89245c171b156a11a4f9697eef0d71dc143

**Comment by aelkiss** [comment]
One concern -- if I use a list of input rights and redirect it to an output file, it's full of stuff like:

```
____________________
^[[1mmdp.39015000952914^[[0m ic
```

I can clean that up but we probably should not be outputting that.

File: bin/bib_rights.pl [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  if (scalar @ids > 1) {
+    printf "%s\n", '_' x 20;
+  }
+}
+
+sub DisplayEntry {
+  my $id    = shift;
+  my $entry = shift;
+
+  print BOLD "$entry->{id}";
```
Comment (aelkiss) [inline bin/bib_rights.pl]: I'd suggest either doing away with the formatting or adding an option to turn it off; perhaps if we provide an input file or detect the output is getting redirected, we could turn off the formatting?

File: bin/bib_rights.pl [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    print BOLD "$id $res->{error}\n";
+    next;
+  }
+  # Show catalog id as separate line if it was a cid query.
+  if ($id eq $res->{cid}) {
+    print BOLD BLUE "$id\n";
+  }
+  DisplayEntry($id, $_) for @{ $res->{entries} };
+  if (scalar @ids > 1) {
+    printf "%s\n", '_' x 20;
```
Comment (aelkiss) [inline bin/bib_rights.pl]: likewise these separators make it slightly more annoying to process the output from this

**Review by aelkiss — CHANGES_REQUESTED** [review]

File: bin/bib_rights.pl (hunk 3 above)
Comment (moseshll) [reply bin/bib_rights.pl]: What would you say to getting rid of the color formatting and going with a YAML-like output format:
```
mdp.0001: pd
000000000:
   mdp.0002: pd
   mdp.0003: ic
```

Comment (aelkiss) [reply bin/bib_rights.pl]: I might be inclined just to have the catalog record # as an additional column rather than try to add structure here?

Comment (moseshll) [reply bin/bib_rights.pl]: See if the most recent push fits the bill. It's on dev-2.

Comment (moseshll) [reply bin/bib_rights.pl]: Apologies, did not see your column suggestion here until after I had tried the YAML-esque format. I think that makes most sense. I'll just use a single space delimiter between htid, cid, attr. Any fields printed with the -v flag will be indented.

Comment (moseshll) [reply bin/bib_rights.pl]: Mods now on dev-2

**Review by aelkiss — APPROVED** [review]
Looks good to me.

**Pull request merged**